tensorflow/core/kernels/depthwise_conv_op_gpu.cu.cc

```diff
@@ -1153,8 +1154,13 @@ __launch_bounds__(1024, 2) void DepthwiseConv2dBackpropFilterGPUKernelNHWCSmall(
       if (filter_depth < in_depth) {
         T val = accum_data[i];
         // Warp-accumulate the pixels of the same depth from the accumulator.
+        int lane_id;
+        asm volatile ("mov.u32 %0, %laneid;" : "=r"(lane_id));
```

Comment:
In the past we've used the threadIdx % 32.  Does that no longer work?

It's kind of a layering violation to put inline assembly here.  If there's no good way to get the lane ID other than inline asm, it should be abstracted away in a function.

Reply:
`cub::LaneId()` does this and I believe cub is a dependency now.

Reply:
I wasn't initially sure of the ordering of treads in thread_idx, but on closer inspection, thread_idx % 32 would give the lane correctly. However, after abstracting the warp reduction into a separate function, it is preferable to not pass a separate thread index / lane id anyway.